src/daemon/dlt-daemon.c

```diff
@@ -835,6 +855,170 @@ int option_file_parser(DltDaemonLocal *daemon_local)
+        /* no context id given, log level is next token */
+        if (pch_next2 == NULL) {
+            log_level = pch_next1;
+        } else {
+            /* context id is given, log level is second to next token */
+            log_level = pch_next2;
+
+            /* next token is token id */
+            strncpy(ctx_id_value, pch_next1, sizeof(ctx_id_value) - 1);
+            if (strlen(ctx_id_value) == 0 || strlen(app_id_value) > DLT_ID_SIZE) {
+                dlt_vlog(LOG_WARNING,
+                         "Invalid ctxid for log settings: app id: %s "
+                         "(skipping line)\n",
+                         app_id_value);
+                continue;
+            }
+
+            ctx_id_value[sizeof(ctx_id_value) - 1] = 0;
+        }
+
+        errno = 0;
+        log_level_value = strtol(log_level, NULL, 10);
+        if (errno != 0 || log_level_value >= DLT_LOG_MAX ||
+            log_level_value <= DLT_LOG_DEFAULT) {
+            dlt_vlog(LOG_WARNING,
+                     "Invalid log level (%i), app id %s, conversion error: %s\n",
+                     log_level_value, app_id_value, strerror(errno));
+            continue;
+        }
+
+        DltDaemonContextLogSettings *settings =
+            dlt_daemon_find_configured_app_id_ctx_id_settings(
+                daemon, app_id_value, NULL);
+
+        if (settings != NULL &&
+            strncmp(settings->ctid, ctx_id_value, DLT_ID_SIZE) == 0) {
+            if (strlen(ctx_id_value) > 0) {
+                dlt_vlog(LOG_WARNING,
+                         "Appid %s with ctxid %s is already configured, skipping "
+                         "duplicated entry\n",
+                         app_id_value, ctx_id_value);
+            } else {
+                dlt_vlog(LOG_WARNING,
+                         "Appid %s is already configured, skipping duplicated entry\n",
+                         app_id_value);
+            }
+
+            continue;
+        }
+
+        /* allocate one more element in the trace load settings */
+        DltDaemonContextLogSettings *tmp =
+            realloc(daemon->app_id_log_level_settings,
+                    (++daemon->num_app_id_log_level_settings) *
+                        sizeof(DltDaemonContextLogSettings));
+
+        if (tmp == NULL) {
+            dlt_log(LOG_CRIT, "Failed to allocate memory for app load settings\n");
+            continue;
+        }
+
+        daemon->app_id_log_level_settings = tmp;
+
+        /* update newly created entry */
+        settings = &daemon->app_id_log_level_settings
+                        [daemon->num_app_id_log_level_settings -1];
+
+        memset(settings, 0, sizeof(DltDaemonContextLogSettings));
+        memcpy(settings->apid, app_id_value, DLT_ID_SIZE);
+        memcpy(settings->ctid, ctx_id_value, DLT_ID_SIZE);
+        settings->log_level = log_level_value;
+
+        /* make sure ids are 0 terminated for printing */
+        char apid_buf[DLT_ID_SIZE + 1] = {0};
```

Comment:
this variable is not used

Reply:
using it for printing now as it was intended